### ISSUE TITLE
Make compiler output logging its own internal task

### ISSUE THREAD

**Issue by cgewecke** [nomiclabs/hardhat#343]
It might be convenient if internal `TASK_COMPILE_COMPILE` consumed a solc output formatter task that other tasks can easily override. Immediate reason for wanting this is that I'm writing something that will inadvertently introduce many warnings. I'd like to suppress them without altering the other logic in that method.

This has wider application - for example Federico Bond wrote some nice error formatting for solc that ended up in [this merged PR](https://github.com/ethereum/solidity/pull/5511). It's not enabled by default and disabled for anyone using the JSON I/O interface. That work could be emulated in JS and have additional features:
+ sorting of warnings and errors
+ warning aggregation (e.g display abi-encoder v2 warning once +  number of times);
+ `--no-warn` 
+ error/warning totals

Basically a plugin like this:
```javascript
usePlugin('pretty-solc');
```
```
npx buidler compile --pretty
```

If there's any appetite here for a PR to decouple the output component, please just lmk and I will send one.

**Comment by alcuadrado** [comment]
Hey @cgewecke!

I really like this idea :) The compile task and its internal tasks are one of the oldest parts of Buidler. They were created when we were still in a prototyping phase. They need to be redesigned soon, before many plugins are created.

Apart from your idea, we have been discussing IRL that they should be redesign with multiple languages in mind. Do you have any other thought around this? Anything else you'd like to be able to do/modify in the compilation process?

**Comment by cgewecke** [comment]
@alcuadrado 
> we have been discussing IRL that they should be redesign with multiple languages in mind

Yes, that makes sense. I don't have any brilliant ideas there unfortunately :) I *think* people have begun mixing Vyper and Solidity in the same project though - for example I believe I saw a project which contains the Uniswap (vyper) contracts as a dep and tests Solidity contract interactions against it. (Where exactly escapes me atm.)

> Anything else you'd like to be able to do/modify in the compilation process?

Yes! I'd like to be able to specify the contents of solc's `outputSelection` parameter. 0xProject's [compiler package](https://sol-compiler.com/) is one design possibility I've seen.

**Comment by alcuadrado** [comment]
> I _think_ people have begun mixing Vyper and Solidity in the same project though

I wasn't expecting this! It will require rethinking the compilation tasks then.

> Yes! I'd like to be able to specify the contents of solc's `outputSelection` parameter. 0xProject's [compiler package](https://sol-compiler.com/) is one design possibility.

We had the same idea :)

Thanks once again for your feedback! Feel free to submit a PR for the change you proposed. It'll be useful, even if we refactor how the compilation tasks work.

**Comment by justinjmoses** [comment]
This would be really great - even just being able to suppress warnings would be 👍
